lib/puppet/pops/evaluator/compare_operator.rb

```diff
@@ -149,4 +155,64 @@ def include_Array(a, b, scope)
   def include_Hash(a, b, scope)
     include?(a.keys, b, scope)
   end
+
+# Implementation of case option matching.
+#
+# This is the type of matching performed in a case option, using == for every type
+# of value except regular expression where a match is performed.
+#
+def is_match? left, right, o, scope
+  if right.is_a?(Regexp)
+    return false unless left.is_a? String
+    matched = right.match(left)
+    set_match_data(matched, scope) # creates or clears ephemeral
+    !!matched # convert to boolean
+  elsif right.is_a?(Puppet::Pops::Types::PAnyType)
+    # right is a type and left is not - check if left is an instance of the given type
+    # (The reverse is not terribly meaningful - computing which of the case options that first produces
+    # an instance of a given type).
+    #
+    @@type_calculator.instance?(right, left)
+  else
```

Comment:
Same here. Call `match_PAnyType(right, left, scope)` instead of repeating its comment and implementation.